### PR TITLE
feat: Improvements to Dashboard

### PR DESCRIPTION
**Pull request by neilLasrado**
- Fixed a missing function in Number Cards
- Made Number Cards & Charts clickable 
Note - Only charts with a group by are clickable. Timeseries & report based charts need to be worked on

dashboard

### CODE REVIEW

File: frappe/public/js/frappe/widgets/chart_widget.js [hunk 1]
```diff
@@ -642,12 +641,42 @@ export default class ChartWidget extends Widget {
 			} else {
 				this.dashboard_chart.update(this.data);
 			}
-
+			this.dashboard_chart.parent.addEventListener("click", (e) => {
+				if ( e.target && e.target.dataset.pointIndex != undefined) {
```
Comment (sahil28297) [inline frappe/public/js/frappe/widgets/chart_widget.js]:
```suggestion
				if (e.target && e.target.dataset && e.target.dataset.pointIndex) {
```

**Review by sahil28297 — CHANGES_REQUESTED** [review]
Also have a look at the sider issues

**Comment by sahil28297** [comment]
@neilLasrado This file `frappe/public/js/frappe/views/dashboard/dashboard_view.js` was intended to be used for this feature https://discuss.erpnext.com/t/new-feature-configurable-dashboard-view/61114. I think we didn't import the part to put it in the views section of sidebar. If we do not wont it, lets merge the PR, otherwise, lets keep all of the files and see if we can get this working.

**Comment by neilLasrado** [comment]
@sahil28297 good find, Let's get this merged for now. We can later import the 'Dashboard; view code if needed.

I strongly feel the code must be commodified into a single file even if we introduce this feature

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**